src/lightkurve/collections.py

```diff
@@ -196,32 +201,13 @@ def stitch(self, corrector_func=lambda x: x.normalize()):
             warnings.filterwarnings("ignore", message=".*already.*")
             lcs = [corrector_func(lc) for lc in self]
 
-        # Address issue #954: ignore incompatible columns with the same name
-        columns_to_remove = set()
-        for col in lcs[0].columns:
-            for lc in lcs[1:]:
-                if col in lc.columns:
-                    if not (
-                        issubclass(lcs[0][col].__class__, lc[col].__class__)
-                        or issubclass(lc[col].__class__, lcs[0][col].__class__)
-                        or lcs[0][col].__class__.info is lc[col].__class__.info
-                    ):
-                        columns_to_remove.add(col)
-                        continue
-
-        if len(columns_to_remove) > 0:
-            warnings.warn(
-                f"The following columns will be excluded from stitching because the column types are incompatible: {columns_to_remove}",
-                LightkurveWarning,
-            )
-            lcs = [lc.copy() for lc in lcs]
-            [
-                lc.remove_columns(columns_to_remove.intersection(lc.columns))
-                for lc in lcs
-            ]
-
         # Need `join_type='inner'` until AstroPy supports masked Quantities
-        return vstack(lcs, join_type="inner", metadata_conflicts="silent")
+        lc_vstack = vstack(lcs, join_type="outer", metadata_conflicts="silent")
+        for col in lc_vstack.columns:
+          if col not in ['time','flux','flux_err']:
+              lc_vstack[col] = lc_vstack[col].value.filled(-999)
```

Comment:
I don't love this approach of using `-999` as a magic number! What about using something like:

```python
lc_vstack[col].fill_value = np.nan
```

in the cases when there is missing data?

Reply:
I definitely agree, we did try that but the problem is that it only works when the column datatype is float, and some of the columns in the lightcurve objects have int datatypes. But hard coding in the -999 didn't feel great either.